README.md

```diff
@@ -5,100 +5,110 @@ Module to generate migrations based on source CSV files
 
-In each source file, we should have Id column required (as unique identifier for row) and all other columns with exact field machine_names. 
+## CSV file structure and contents
 
-Need to take care about next things:
-* Language field should have language code in source csv, not labels
+We should use next ways of organizing source files:
+* Source CSV files should next filename`{entity_type}-{bundle}.csv` or `{entity_type}.csv`.
+
+  You can see some examples in /example folder of this module, like `node-basic_page.csv`, `taxonomy_term-category.csv`, `user.csv`, etc.
+
+**Main idea here - to identify target Entity type and Bundle for source file.**
+
+There are some rules for structure and content of source csv file:
+* Each source file should have first column filled with Ids (it will be used as unique identifier for source row)
+* All other columns should be named with exact field's machine name.
+* For boolean fields you can use "1/0" or "true/false" values.
+* For list fields, we should have key values in source csv, not labels.
 * Fields with multiple values should have strict delimiter. `|` for example.
-* For list fields, we should have keys in source csv, not labels
-* Date fields should have strict date format everywhere.
-* Sometimes, field consists of several values, need to support that too. We can use `/` separator in column name for these cases.
-Example of field types to support:
-  - **formatted text** (we have format and value)
-  - **link**
-  - **datetime_range**
-  - **address**
-  - **price**. 
- 
-* For file migration, we could have 2 possible cases:
-  - we have absolute filepath in csv -> then we use file_copy process plugin
-  - we have URLs in csv -> we use download plugin
 
-   and we could use our plugin https://www.drupal.org/project/migrate_plus/issues/3113394 here too
+  This delimiter can be specified by **values_delimiter** option in drush command.
+* Date fields should have same date format everywhere in sources.
 
-**TODO:** What other specific field types should we support?
+  This date format can be specified by **date_format** option in drush command.
 
-------------
-## Reference fields support
+## Complex fields with multiple properties.
 
-**References is most complicated part here, as we should have migration dependencies for them**
+For complex fields, like Formatter text, Link, Datetime Range, Address, Price, we could have several properties in source file.
+In this case, you can use `/` separator in column name for these cases.
 
-We could handle it next way:
-1. We know, what sources do we have.
-2. We know field machine name from csv header, so from field definition we could know if it is reference field and what entity and bundle is the target
+For example, `body/value`, `body/format`, `price/number`, `price/currency_code`, etc.
 
-So, if we also have a source for this entity and bundle => then **we have migration dependency here**
+Example of supported complex field types:
```

Comment:
Would be great to have real exemple for each field

Reply:
reworked examples